source/lac/dynamic_sparsity_pattern.cc

```diff
@@ -363,6 +363,11 @@ DynamicSparsityPattern::exists(const size_type i, const size_type j) const
   if (!have_entries)
     return false;
 
+  // Avoid a segmentation fault in below code if the row index happens to
+  // not be present in the IndexSet rowset:
+  if (rowset.size() != 0 && !rowset.is_element(i))
+    return false;
```

Comment:
@bangerth I have left the Assert that you strongly insist on in place. I suggest to still merge the 4 lines in order to avoid a segmentation fault.

Reply:
In other words, if you make it past the assertion, then `rowset.is_element(i)==true` and the condition here can never be satisfied. Do I understand right then that you're only trying to avoid the segfault that would happen in release mode but that could never happen in debug mode?

Reply:
Yes.

Reply:
Then maybe write it like this:
```
  // The assertion above guarantees that we are dealing with an 'i' for which we store rows.
  // But if one runs the code below in release mode, one would get a segfault, so be friendly
  // and avoid this:
  if (! (...condition in assertion...))
    return 0;
```
Though I'm not sure we should really be friendly to people who run in release mode and complain about segfaults -- they get what they deserve.

Reply:
@bangerth Since when are we so mean to users?